packages/flutter/test/services/delta_text_input_test.dart

```diff
@@ -0,0 +1,174 @@
+      () async {
+        // Assemble a TextInputConnection so we can verify its change in state.
+        final FakeDeltaTextInputClient client = FakeDeltaTextInputClient(TextEditingValue.empty);
+        const TextInputConfiguration configuration = TextInputConfiguration(enableDeltaModel: true);
+        TextInput.attach(client, configuration);
+        expect(client.configuration.enableDeltaModel, true);
+
+        expect(client.latestMethodCall, isEmpty);
+
+        const String jsonDelta = '{'
+            '"oldText": "",'
+            ' "deltaText": "let there be text",'
+            ' "deltaStart": 0,'
+            ' "deltaEnd": 0,'
+            ' "selectionBase": 17,'
+            ' "selectionExtent": 17,'
+            ' "selectionAffinity" : "TextAffinity.downstream" ,'
+            ' "selectionIsDirectional": false,'
+            ' "composingBase": -1,'
+            ' "composingExtent": -1}';
+
+        // Send updateEditingValueWithDeltas message.
+        final ByteData? messageBytes = const JSONMessageCodec().encodeMessage(<String, dynamic>{
+          'args': <dynamic>[
+            1,
+            jsonDecode('{"deltas": [$jsonDelta]}'),
+          ],
+          'method': 'TextInputClient.updateEditingStateWithDeltas',
+        });
+        await ServicesBinding.instance!.defaultBinaryMessenger.handlePlatformMessage(
+          'flutter/textinput',
+          messageBytes,
+              (ByteData? _) {},
+        );
+
+        expect(client.latestMethodCall, 'updateEditingValueWithDeltas');
+      },
+    );
+  });
+}
+
+class FakeDeltaTextInputClient implements DeltaTextInputClient {
+  FakeDeltaTextInputClient(this.currentTextEditingValue);
+
+  String latestMethodCall = '';
+
+  @override
+  TextEditingValue currentTextEditingValue;
+
+  @override
+  AutofillScope? get currentAutofillScope => null;
+
+  @override
+  void performAction(TextInputAction action) {
+    latestMethodCall = 'performAction';
+  }
+
+  @override
+  void performPrivateCommand(String action, Map<String, dynamic> data) {
+    latestMethodCall = 'performPrivateCommand';
+  }
+
+  @override
+  void updateEditingValue(TextEditingValue value) {
+    latestMethodCall = 'updateEditingValue';
+  }
+
+  @override
+  void updateEditingValueWithDeltas(List<TextEditingDelta> textEditingDeltas) {
+    latestMethodCall = 'updateEditingValueWithDeltas';
+  }
+
+  @override
+  void updateFloatingCursor(RawFloatingCursorPoint point) {
+    latestMethodCall = 'updateFloatingCursor';
+  }
+
+  @override
+  void connectionClosed() {
+    latestMethodCall = 'connectionClosed';
+  }
+
+  @override
+  void showAutocorrectionPromptRect(int start, int end) {
+    latestMethodCall = 'showAutocorrectionPromptRect';
+  }
+
+  TextInputConfiguration get configuration => const TextInputConfiguration(enableDeltaModel: true);
+}
+
+class FakeTextChannel implements MethodChannel {
```

Comment:
Nit: Is this identical to the one in text_input_test.dart?  Could the code be shared somehow?  Maybe a text_input_utils.dart file or something.

Reply:
Yeah it's shared by `autofill_test.dart` and `text_input_test.dart` so i'll make it a common file.